core/tests/unittests/test_feature_selection.py

```diff
@@ -0,0 +1,89 @@
+from autogluon.core.utils.feature_selection import *
+from autogluon.core.utils.utils import unevaluated_fi_df_template
+import numpy as np
+from numpy.core.fromnumeric import sort
+import pandas as pd
+
+
+def evaluated_fi_df_template(features, importance=None, n=None):
+    rng = np.random.default_rng(0)
+    importance_df = pd.DataFrame({'name': features})
+    importance_df['importance'] = rng.standard_normal(len(features)) if importance is None else importance
+    importance_df['stddev'] = rng.standard_normal(len(features))
+    importance_df['p_value'] = None
+    importance_df['n'] = 5 if n is None else n
+    importance_df.set_index('name', inplace=True)
+    importance_df.index.name = None
+    return importance_df
+
+
+def test_add_noise_column():
+    X = None
+    args = {'prefix': 'noise_prefix', 'rng': np.random.default_rng(0), 'count': 2}
+    assert add_noise_column(X, **args) is None
+    X = pd.DataFrame({'a': [1, 2]})
+    feature_metadata = FeatureMetadata.from_df(X)
+    X_noised = add_noise_column(X, **args, feature_metadata=feature_metadata)
+    expected_features = X.columns.tolist() + ['noise_prefix_1', 'noise_prefix_2']
+    assert expected_features == X_noised.columns.tolist()
+    assert expected_features == feature_metadata.get_features()
+
+
+def test_merge_importance_dfs():
+    # test first time
+    features = ['a', 'b', 'c', 'd', 'e']
+    prev_fit_estimates = set()
+    df1, df2 = None, unevaluated_fi_df_template(features)
+    assert merge_importance_dfs(df1, df2, prev_fit_estimates) is df2
+    # test subsequent time where no estimates are from previous runs
+    df1 = evaluated_fi_df_template(features, importance=[0.2, 0.2, None, 1., None], n=[10, 5, 0, 5, 0])
+    df2 = evaluated_fi_df_template(features, importance=[-0.1, -0.1, 0.1, None, None], n=[5, 10, 10, 0, 0])
+    result_df = merge_importance_dfs(df1, df2, prev_fit_estimates)
+    assert [score if score == score else None for score in result_df['importance'].tolist()] == [0., 0.1, 0.1, 1., None]
+    assert result_df['n'].tolist() == [15, 15, 10, 5, 0]
+    # test subsequent time where some estimates are from previous runs
+    prev_fit_estimates = set(['a'])
+    result_df = merge_importance_dfs(df1, df2, prev_fit_estimates)
+    assert [score if score == score else None for score in result_df['importance'].tolist()] == [-0.1, 0., 0.1, 1., None]
+    assert result_df['n'].tolist() == [5, 15, 10, 5, 0]
+    assert prev_fit_estimates == set()
+
+
+def test_sort_features_by_priority():
```

Comment:
Consider splitting the tests below into a separate test cases each with fixtures to reduce duplication. Rationale: this will create a finer failure picture when multiple things get broken.